src/main/java/com/google/cloud/tools/sdk/CloudSdkCliExecutor.java

```diff
@@ -0,0 +1,60 @@
+package com.google.cloud.tools.sdk;
+
+import com.google.cloud.tools.internal.process.AsynchronousProcessRunnerRunner;
+import com.google.cloud.tools.internal.process.ProcessRunner;
+import com.google.cloud.tools.internal.process.ProcessRunnerException;
+import com.google.cloud.tools.internal.process.SimpleProcessRunner;
+import com.google.cloud.tools.internal.sdk.CloudSdk;
+import com.google.cloud.tools.internal.sdk.CloudSdkCommandFactory;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.Future;
+
+/**
+ * Created by appu on 3/30/16.
+ */
+public class CloudSdkCliExecutor implements CloudSdkExecutor {
```

Comment:
I think this is too many abstraction layers just to make system calls. What advantages are there for this vs ProcessCaller?